src/coord/src/coord.rs

```diff
@@ -3417,6 +3396,16 @@ where
         None
     };
 
+    let (internal_cmd_tx, internal_cmd_rx) = futures::channel::mpsc::unbounded();
+    let (ts_tx, ts_rx) = std::sync::mpsc::channel();
+    let mut timestamper = Timestamper::new(&timestamp_config, internal_cmd_tx.clone(), ts_rx);
+    let executor = Handle::current();
+    let _timestamper_thread = thread::spawn(move || {
+        let _executor_guard = executor.enter();
+        timestamper.update()
+    })
+    .join_on_drop();
```

Comment:
chatted with @benesch about this and the summary is -- we can't do `join_on_drop` here because this means "when this object `_timestamper_thread` is dropped, please wait for the spawned thread to exit" which is not gonna happen. Instead, we need to find a way to push the return value from thread::spawn into coord::serve so that it can be appropriately dropped when the coordinator itself is ready to drop

Reply:
Just to clarify: the `join_on_drop` is great here! It's just that the resulting `timestamper_thread: JoinOnDropHandle<()>` needs to get stored as a field on `Coordinator`.

Reply:
yes thank you! that is what i wanted to type, but entirely failed at :D